docs/range-proof-protocol.md

```diff
@@ -1,190 +1,315 @@
-The `range_proof` module contains API for producing a range proof for a given integer value.
+The `aggregated_range_proof` module contains an API for producing a proof that multiple integer values are within a certain range.
 
-Range proof protocol
-====================
+Aggregated proof protocol
+=========================
 
-This is a documentation for the internal implementation of a range proof. 
-You may find the introduction to all the pieces of the protocol in the [notes](../notes/index.html) module.
+This is a documentation for the internal implementation of the aggregated range proof. You may find the introduction to all the pieces of the range proof protocol in the [notes](../notes/index.html) module.
 
-The range proof is a zero-knowledge proof of the following relation
+The aggregated range proof is a zero-knowledge proof of the following relation:
 \\[
 \operatorname{ZK-PK}\left\\{
-  v \in {\mathbb Z\_p}
-  : v \in [0, 2^n)
+  v_{(j)} \in {\mathbb Z\_p}
+  : v_{(j)} \in [0, 2^n)
 \right\\}
-\\] where \\(n = 2^{k}\\) is a power of \\(2\\).
+\\] 
 
-Prover’s algorithm
+for \\(m\\) values \\(v_{(j)}\\), where \\(n\\) and \\(m\\) are both a power of \\(2\\).
+
+API overview
+------------
+
+The prover is composed of \\(m\\) individual parties and one dealer that coordinates messages and provides challenges.
+
+First, parties are instantiated with their secret values and the range for the range proof. The dealer assigns a unique index \\(j\\) to each party.
+
+Next, each party commits to its secret value, bits, and blinding factors via \\(V_{(j)}, A_{(j)}, S_{(j)}\\) as `ValueCommitment` and sends `ValueCommitment` to the dealer. The dealer receives and combines all of the `ValueCommitment`s from all of the parties, generates challenges \\(y,z\\) as `ValueChallenge`, and sends `ValueChallenge` to all of the parties.
```

Comment:
This could be shortened a little bit by saying, e.g.,
```
... blinding factors via \\( ... \\) and sends this data in a `ValueCommitment` message to ...
```
which avoids saying the message name twice, and makes it clear that it is a message.

Reply:
👍